### PR TITLE
Issue #57, initial commit for review of RecordNotUnique error handling i...

### PR DESCRIPTION
**Pull request by arielspear**
...n concept controller.

Me  and @shawnacscott are unsure about where/what the record's location is.

### CODE REVIEW

File: app/controllers/concepts_controller.rb [hunk 1]
```diff
@@ -1,4 +1,13 @@
+#It shouldn't be possible to create two concepts with the same name.
+#Create a unique index, validate_uniqueness_of :name,
+#and add a RecordExistsError handler in the ConceptsController.
+#should flash that record exists, including it's name and location
+#and redirect to creating a new page, using information
+#that allready was entered
+
 class ConceptsController < ApplicationController
+  validate_uniqueness_of :name
```
Comment (ErinCall) [inline app/controllers/concepts_controller.rb]: This should actually go in the Concept model.

File: app/controllers/concepts_controller.rb [hunk 2]
```diff
@@ -8,11 +17,14 @@ def new
...
-      redirect_to @concept
-    else
+    begin
+      @concept = Concept.new(concept_params)
+      if @concept.save
+        flash[:notice] = "Your concept has been added."
+        redirect_to @concept
+      end
+    rescue ActiveRecord::RecordNotUnique => e
+      flash[:notice] = "#{@concept} already exists."
```
Comment (ErinCall) [inline app/controllers/concepts_controller.rb]: I think what you want here is `"#{@concept.name} already exists"`, or maybe `"There is already a concept called #{@concept.name}"`.

**Comment by ErinCall** [comment]
I think your general approach is good. You could get the conflicting concept with something like

``` Ruby
other_concept = Concept.find_by(name: @concept.name)
```

But there isn't much need to do it.

**Comment by arielspear** [comment]
That does help us with a problem. However, I think we wanted to know more about what you meant when you asked us to list the location of the record.

(The name thing is very helpful though and we knew we didn't quite have that part right yet, thank you.)

**Comment by ErinCall** [comment]
Oh then you would say `concept_url(other_concept)` and that would return something like `/concepts/5`

**Comment by arielspear** [comment]
Thank you, I think this helps a lot. I'll ask @shawnacscott if that was all we needed when I see her at class tonight.

**Comment by arielspear** [comment]
Will {concept_url(other_concept)} provide a link, or only print the url?

**Comment by ErinCall** [comment]
It will only generate a URL. If you want a link you should use `link_to` as well.

**Comment by ErinCall** [comment]
The code that is here looks right! However, there're two things missing:
- Tests
- There needs to be a schema migration adding a unique index to the concept table's name column.

**Comment by shawnacscott** [comment]
@pythonchelle @jeffweiss @keturn Would any of ya'll mind closing this PR? It shows up forever for me in my GH pull request view. Thank yoooou <3

**Pull request closed without merging**